### PR TITLE
Test controllers

### PR DESCRIPTION
**Pull request by bamsi**
In this pull request, I have implemented the following:
- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
    - If response status was correct.
    - If a correct template was rendered.
    - If the response body includes correct placeholder text.
  Your code review is appreciated!

### CODE REVIEW

File: spec/controllers/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
```
Comment (medaminedev66) [inline spec/controllers/posts_spec.rb]: - [ ] Following the project requirements, your project should Have `users_spec.rb` and `posts_spec.rb` in `spec/requests/` directory.

File: spec/controllers/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+RSpec.describe PostsController, type: :controller do
```
Comment (medaminedev66) [inline spec/controllers/posts_spec.rb]: - [ ] Following the project requirements, you should use `type: request` as shown in the example code below OR use `config.infer_spec_type_from_file_location!` in the Rspec config file (in that case just placing a file in the `spec/requests` dir will be enough).
```ruby
RSpec.describe "Users", type: :request do
   ...
 end
```

File: spec/controllers/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  describe 'GET #show' do
+    it 'returns a 200 status code' do
+      subject { get :show, params: { user_id: @user.id, id: @post.id } }
+      expect(subject).to have_http_status(200)
+    end
+
+    it 'render show template' do
+      subject { get :show, params: { user_id: @user.id, id: @post.id } }
+      expect(subject).to render_template('show')
```
Comment (medaminedev66) [inline spec/controllers/posts_spec.rb]: - [ ] Following the project requirements, make sure that for each action you check:
- If response status was correct.  
- If a correct template was rendered. (Something like ``` expect(response).to render_template(:show)```)
- If the response body includes correct placeholder text. (Something like ```expect(response.body).to include("Welcome to user index page")```)

All your tests fail, please work on correcting them:

image

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @bamsi  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional 
- You have a descriptive PR 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      subject { get :index, params: { user_id: @user.id } }
+      expect(subject).to render_template('index')
+    end
+
+    it 'the response body includes correct placeholder' do
+      subject { get :index, params: { user_id: @user.id } }
+      expect(subject.body).to include('Here is a list of posts for a given user')
+    end
+  end
+
```
Comment (phelian23) [inline spec/requests/posts_spec.rb]: - [ ] Good job working on the changes requested by the previous reviewer, however, your test is still failing.
Capture

- [ ] Check your models. the validate method should be `validates` and not `Validate`
- [ ] Also in your migration files, the references are not set up correctly

**Review by phelian23 — CHANGES_REQUESTED** [review]
Hi @bamsi,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @bamsi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## To Highlight :clap: 
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Good job working on the changes requested by the previous reviewer, however, you're still having 12 failed test, please work on correcting them:




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @bamsi , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

**All the tests are running successfully** 🎉 


Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**